src/plugins/cache.js

```diff
@@ -107,6 +127,21 @@ export default class Cache {
 
   async fetchManagers (...managers: Manager[]): Promise<Plugin[]> {
     let plugins = []
+    if (this.cache.node_version !== process.version) {
+      let lock = new Lock(this.out)
+      let handled = true
+
+      let downgrade = await lock.upgrade()
+      for (let manager of managers) {
+        handled = await manager.handleNodeVersionChange() && handled
```

Comment:
@dickeyxxx ignored failing rebuilds in https://github.com/heroku/cli-engine/pull/211/commits/0a19536ab5b534fe7a1ab735b0911920d67ff6f5